pkg/virt-controller/services/template.go

```diff
@@ -1251,7 +1247,20 @@ func wrapGuestAgentPingWithVirtProbe(vmi *v1.VirtualMachineInstance, probe *k8sv
 	return
 }
 
-func alignPodMultiCategorySecurity(pod *k8sv1.Pod, selinuxType string, dockerSELinuxMCSWorkaround bool) {
+func alignPodMultiCategorySecurity(pod *k8sv1.Pod, vmi *v1.VirtualMachineInstance, selinuxType string, dockerSELinuxMCSWorkaround bool) {
+	if selinuxType == "" {
+		if util.HasHugePages(vmi) || util.IsVMIVirtiofsEnabled(vmi) {
```

Comment:
@jean-edouard do we need to include passt here as well?

Reply:
I'm not 100% sure what tests we have for `passt`. Perhaps, will not work correctly without selinux policy, but will not crash qemu...
I think we should include it here as long we have it in our custom policy. wdyt?

Reply:
It is possible we consume a new enough `container_t` policy that allows what we need.

Reply:
@xpivarc good point. I see plenty of passt-related functional tests in the suite.
I think we are good to go then. Once this PR merge, I will document the fact that, on SELinux-enabled nodes, KubeVirt requires container-selinux v2.170.0 or above (what CI currently uses).
That version is a year old at this point, so it shouldn't be a problem.
It would however be nice to know which permission was assumed to be needed for passt so we can ensure we're not missing anything.

Reply:
Sounds good! But if this is the case, let's get rid of passt related entries in our policy too. (In a follow-up)

Reply:
/hold
I checked container_t is not allowed to mount tmpfs, which is the permission that was added to the virt-launcher policy.
We need to get to the bottom of it before merging this.

Reply:
/hold cancel
Although container_t doesn't have an explicit rule for allowing tmpfs mounts, it somehow allows what passt needs.
However, the custom virt-launcher policy still needs an added permission to be compatible with passt.
Even though this PR runs passt VMIs as container_t, passt+virtiofs/hugepages VMIs would still run under the custom policy. Therefore the permission cannot be removed.
Functional tests were added for such cases.
@acardace PTAL!